ext/ddtrace_profiling_native_extension/profiling.c

```diff
@@ -111,14 +112,28 @@ static VALUE _native_trigger_holding_the_gvl_signal_handler_on(DDTRACE_UNUSED VA
 
   pthread_mutex_lock(&holding_the_gvl_signal_handler_mutex);
 
-  for (int tries = 0; holding_the_gvl_signal_handler_result[0] == Qfalse && tries < 100; tries++) {
+  for (int tries = 0; holding_the_gvl_signal_handler_result[0] == Qfalse && tries < 500; tries++) {
```

Comment:
Yes! This spec is inherently a bit fragile since it relies on multiple threads and some communication between them. I've increased the tries from 100 (that is 100 x 10 => 1 second) to 500 (500 x 10 => 5 seconds) to avoid any potential flakiness caused by the test machine being slow.

I've documented this in f66bc7d3af26bcc0805e1eb0eaceade13967caf8